src/coreclr/vm/ilstubcache.h

```diff
@@ -112,49 +91,29 @@ class ILStubCache : private CClosedHashBase
         SigTypeContext *pTypeContext,
         AllocMemTracker* pamTracker);
 
-    // *** OVERRIDES FOR CClosedHashBase ***/
-
-    //*****************************************************************************
-    // Hash is called with a pointer to an element in the table.  You must override
-    // this method and provide a hash algorithm for your element type.
-    //*****************************************************************************
-    virtual unsigned int Hash(             // The key value.
-        void const*  pData);                // Raw data to hash.
-
-    //*****************************************************************************
-    // Compare is used in the typical memcmp way, 0 is eqaulity, -1/1 indicate
-    // direction of miscompare.  In this system everything is always equal or not.
-    //*****************************************************************************
-    virtual unsigned int Compare(          // 0, -1, or 1.
-        void const*  pData,                 // Raw key data on lookup.
-        BYTE*        pElement);             // The element to compare data against.
-
-    //*****************************************************************************
-    // Return true if the element is free to be used.
-    //*****************************************************************************
-    virtual ELEMENTSTATUS Status(           // The status of the entry.
-        BYTE*        pElement);             // The element to check.
-
-    //*****************************************************************************
-    // Sets the status of the given element.
-    //*****************************************************************************
-    virtual void SetStatus(
-        BYTE*         pElement,             // The element to set status for.
-        ELEMENTSTATUS eStatus);             // New status.
-
-    //*****************************************************************************
-    // Returns the internal key value for an element.
-    //*****************************************************************************
-    virtual void* GetKey(                   // The data to hash on.
-        BYTE*        pElement);             // The element to return data ptr for.
+private: // Inner classes
+    struct ILStubCacheEntry
+    {
+        MethodDesc *m_pMethodDesc;
+        ILStubHashBlob *m_pBlob;
+    };
+
+    class ILStubCacheTraits : public DeleteElementsOnDestructSHashTraits<DefaultSHashTraits<ILStubCacheEntry *>>
```

Comment:
Small element types like `ILStubCacheEntry` are typically stored in SHash inplace for better locality and efficiency. Consider changing the element type to `ILStubCacheEntry`.